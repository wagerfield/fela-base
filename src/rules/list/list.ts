export interface ListRuleOptions {
  padding?: any
  margin?: any
  style?: any
}

export default ({
  style = "none",
  padding = 0,
  margin = 0
}: ListRuleOptions = {}) => ({
  listStyle: style,
  padding,
  margin
})
