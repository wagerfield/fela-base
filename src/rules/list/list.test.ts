import list from "./list"

describe("list(props)", () => {
  it("renders expected snapshots", () => {
    expect(list()).toMatchSnapshot()
    expect(
      list({
        style: "square",
        padding: 10
      })
    ).toMatchSnapshot()
  })
})
