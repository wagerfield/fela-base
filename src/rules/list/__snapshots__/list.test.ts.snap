// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list(props) renders expected snapshots 1`] = `
Object {
  "listStyle": "none",
  "margin": 0,
  "padding": 0,
}
`;

exports[`list(props) renders expected snapshots 2`] = `
Object {
  "listStyle": "square",
  "margin": 0,
  "padding": 10,
}
`;
